.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,83 @@
+name: DCE deploy to environment
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+        description: "The environment to deploy to (e.g. nonprod, prod)"
+        default: "nonprod"
+      branch:
+        required: false
+        type: string
+        default: master
+      release:
+        required: false
+        type: string
+        default: none
+      go_version:
+        required: false
+        type: string
+        default: "1.21.6"
+      terraform_version:
+        required: false
+        type: string
+        default: "1.7.4"
+      version_tag:
+        required: false
+        type: string
+        default: master # pr-#, nonprod, or vX.Y.Z
+      namespace:
+        required: false
+        type: string
+        default: nonprod # pr-#, nonprod, or prod
+    secrets:
+      AWS_DEFAULT_REGION:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      DCE_NONPROD_HOSTNAME:
+        required: true
+      NOTIFY_EMAIL:
+        required: true
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Download bin artifact
+        uses: actions/download-artifact@v4
```

Comment:
Does this work? One potential scenario where it might not is there are 2 prs, both run successfully but pr-2 runs afterwards, but pr-1 is merged. Wouldn't pr-2's artifacts be available.